src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -260,20 +260,15 @@ public function configureOptions(OptionsResolver $resolver)
             // Harden against NULL values (like in EntityType and ModelType)
             $choices = null !== $options['choices'] ? $options['choices'] : array();
 
-            // BC when choices are in the keys, not in the values
-            if (!$options['choices_as_values']) {
-                return $choiceListFactory->createListFromFlippedChoices($choices, $options['choice_value'], false);
-            }
-
             return $choiceListFactory->createListFromChoices($choices, $options['choice_value']);
         };
 
         $choicesAsValuesNormalizer = function (Options $options, $choicesAsValues) {
-            if (true !== $choicesAsValues) {
-                @trigger_error('The value "false" for the "choices_as_values" option is deprecated since version 2.8 and will not be supported anymore in 3.0. Set this option to "true" and flip the contents of the "choices" option instead.', E_USER_DEPRECATED);
+            if (null !== $choicesAsValues) {
+                @trigger_error('The "choices_as_values" option is ignored since version 3.0 and will be removed in 4.0.', E_USER_DEPRECATED);
```

Comment:
@stof suggested to trigger this deprecation later in a 3.x release to be able to have bundles supporting 2.x/3.x that do not trigger deprecations